server/src/com/cloud/network/router/VirtualNetworkApplianceManagerImpl.java

```diff
@@ -916,6 +916,9 @@ protected void updateSite2SiteVpnConnectionState(final List<DomainRouterVO> rout
                 }
                 continue;
             }
+            if (router.getIsRedundantRouter() && router.getRedundantState() != RedundantState.MASTER){
```

Comment:
ok @pdube so getConnectionsForRouter() returns vpn connections based on the vpcid of the router. As both the backup and master router have the same vpcid assigned it makes sense to first exclude backup routers and then to exclude non-running routers. 

I forced pushed the change. @pdube could you test?